libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -2,111 +2,238 @@ import * as grpc from '@grpc/grpc-js';
           })
         );
+
+    if (config.cache === 'lru') {
+      this._cacheEnabled = true;
+      this._cache = new LRU({ maxSize: config.maxCacheSize || DEFAULT_MAX_CACHE_SIZE, sizeCalculation: () => 1 });
+      this.streamConnection = this.connectStream();
+    }
   }
 
   async resolveBoolean(
     flagKey: string,
     context: EvaluationContext,
     logger: Logger
   ): Promise<ResolutionDetails<boolean>> {
-    const { response } = await this.client.resolveBoolean({
-      flagKey,
-      context: this.convertContext(context, logger),
-    }).then(this.onFulfilled, this.onRejected);;
-    return {
-      value: response.value,
-      reason: response.reason,
-      variant: response.variant,
-    };
+    return this.resolve(this._client.resolveBoolean, flagKey, context, logger);
   }
 
-  async resolveString(
-    flagKey: string,
-    context: EvaluationContext,
-    logger: Logger
-  ): Promise<ResolutionDetails<string>> {
-    const { response } = await this.client.resolveString({
-      flagKey,
-      context: this.convertContext(context, logger),
-    }).then(this.onFulfilled, this.onRejected);
-    return {
-      value: response.value,
-      reason: response.reason,
-      variant: response.variant,
-    }
+  async resolveString(flagKey: string, context: EvaluationContext, logger: Logger): Promise<ResolutionDetails<string>> {
+    return this.resolve(this._client.resolveString, flagKey, context, logger);
   }
 
-  async resolveNumber(
-    flagKey: string,
-    context: EvaluationContext,
-    logger: Logger
-  ): Promise<ResolutionDetails<number>> {
-    const { response } = await this.client.resolveFloat({
-      flagKey,
-      context: this.convertContext(context, logger),
-    }).then(this.onFulfilled, this.onRejected);
-    return {
-      value: response.value,
-      reason: response.reason,
-      variant: response.variant,
-    };
+  async resolveNumber(flagKey: string, context: EvaluationContext, logger: Logger): Promise<ResolutionDetails<number>> {
+    return this.resolve(this._client.resolveFloat, flagKey, context, logger);
   }
 
   async resolveObject<T extends JsonValue>(
     flagKey: string,
     context: EvaluationContext,
     logger: Logger
   ): Promise<ResolutionDetails<T>> {
-    const { response } = await this.client.resolveObject({
-      flagKey,
-      context: this.convertContext(context, logger),
-    }).then(this.onFulfilled, this.onRejected);
-    if (response.value !== undefined) {
-      return {
-        value: Struct.toJson(response.value) as T,
-        reason: response.reason,
-        variant: response.variant,
-      };
+    return this.resolve(this._client.resolveObject, flagKey, context, logger, this.objectParser);
+  }
+
+  private connectStream(): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      this.logger?.info(`${FlagdProvider.name}: connecting stream, attempt ${this._streamConnectAttempt}...`);
```

Comment:
Should this be info level?

Reply:
I debated this - I though yes since it's something that would happen infrequently. I could be convinced otherwise.

Reply:
I got rid of a lot of `info`s, I think you're right.

I only kept the connection establishment to info.